osf_tests/factories.py

```diff
@@ -1142,12 +1141,15 @@ def _create(cls, *args, **kwargs):
 
         registration.registered_schema.clear()
         registration.registered_schema.add(schema)
```

Comment:
This creates weirdness of maybe having responses for multiple schemas on the registration -- technically possible, but not really what we want. It will especially cause issues with the `create_from_previous_responses` case, since the SchemaResponseBlocks we inherit won't match the schema here.

We should just grab the schema from registration, and possibly accept a `schema` argument to pass to the RegistrationFactory in the case where `registration` isn't provided?